src/SarifCli/AnalyzeCommand.cs

```diff
@@ -0,0 +1,148 @@
+            context.SchemaFilePath = options.SchemaFilePath;
+            return context;
+        }
+
+        protected override void AnalyzeTarget(IEnumerable<ISkimmer<SarifValidationContext>> skimmers, SarifValidationContext context, HashSet<string> disabledSkimmers)
+        {
+            // The base class knows how to invoke the skimmers that implement smart validation...
+            base.AnalyzeTarget(skimmers, context, disabledSkimmers);
+
+            // ... but it doesn't know how to invoke schema validation, which has its own set of rules,
+            // so we do that ourselves.
+            Validate(context.TargetUri.LocalPath, context.SchemaFilePath, context.Logger);
+        }
+
+        private void Validate(string instanceFilePath, string schemaFilePath, IAnalysisLogger logger)
+        {
+            try
+            {
+                string instanceText = File.ReadAllText(instanceFilePath);
+                PerformSchemaValidation(instanceText, instanceFilePath, schemaFilePath, logger);
+            }
+            catch (JsonSyntaxException ex)
+            {
+                ReportResult(ex.Result, logger);
+            }
+            catch (SchemaValidationException ex)
+            {
+                ReportInvalidSchemaErrors(ex, schemaFilePath, logger);
+            }
+            catch (Exception ex)
+            {
+                LogToolNotification(logger, ex.Message, NotificationLevel.Error, ex);
+            }
+        }
+
+        private void PerformSchemaValidation(
+            string instanceText,
+            string instanceFilePath,
+            string schemaFilePath,
+            IAnalysisLogger logger)
+        {
+            string schemaText = File.ReadAllText(schemaFilePath);
+            JsonSchema schema = SchemaReader.ReadSchema(schemaText, schemaFilePath);
+
+            var validator = new Validator(schema);
+            Result[] results = validator.Validate(instanceText, instanceFilePath);
+
+            ReportResults(results, logger);
+        }
+
+        private static void ReportInvalidSchemaErrors(
+            SchemaValidationException ex,
+            string schemaFile,
+            IAnalysisLogger logger)
+        {
+            foreach (Result result in ex.Results)
+            {
+                result.SetAnalysisTargetUri(schemaFile);
+                ReportResult(result, logger);
+            }
+        }
+
+        private static void ReportResults(
+            Result[] results,
+            IAnalysisLogger logger)
+        {
+            foreach (Result result in results)
+            {
+                ReportResult(result, logger);
+            }
+        }
+
+        private static void ReportResult(
+            Result result,
+            IAnalysisLogger logger)
+        {
+            Rule rule = Json.Schema.Sarif.RuleFactory.GetRuleFromRuleId(result.RuleId);
+            logger.Log(rule, result);
+        }
+
+        private static void LogToolNotification(
```

Comment:
Fixed.

---
In reply to: [77238913](https://github.com/Microsoft/sarif-sdk/pull/501#discussion_r77238913) [](ancestors = 77238913,77238560)